### ISSUE TITLE
Suggested policy: suggest @{^CAPTURE}

### ISSUE THREAD

**Issue by epa** [Perl-Critic/Perl-Critic#927]
Lots of Perl code looks like

```
if (/(some) (regexp) (with) (captures)/) {
    my @stuff = ($1, $2, $3, $4);
    ...
}
```

This could nowadays be expressed better as

```
if (/(some) (regexp) (with) (captures)/) {
    my @stuff = @{^CAPTURE};
    ...
}
```

Perlcritic should suggest using `@{^CAPTURE}` instead of a long list of `$1, $2, $3...`.  I would do so whenever the assignment immediately follows a regexp match and has the same number of capture variables as there were capturing groups in the regexp, and there are at least three.

**Comment by petdance** [comment]
What version of Perl did `@{^CAPTURE}` show up?

**Comment by epa** [comment]
It was added in 5.26.  https://github.com/Perl/perl5/issues/13412

**Comment by petdance** [comment]
I don't think we want to add that as a core policy for something that new. I think it would be great as an add-on policy.

**Comment by epa** [comment]
Since more time has passed since the release of perl 5.26 (on May 30 2017), could I ask you to reconsider in principle adding this as a core policy?

If it's still too soon, how old would a perl feature need to be before perlcritic can suggest using it?

**Comment by petdance** [comment]
Leaving aside the question of how current a version of Perl should be, do you have any code to support such a feature?
